tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -1860,17 +1881,15 @@ public function testIncrementOnExistingModelCallsQueryAndSetsAttribute()
         $model->syncOriginalAttribute('id');
         $model->foo = 2;
 
-        $model->shouldReceive('newQueryWithoutRelationships')->andReturn($query = m::mock(stdClass::class));
-        $query->shouldReceive('where')->andReturn($query);
-        $query->shouldReceive('increment');
+        $model->shouldReceive('newQueryWithoutRelationships')->andReturn($this->mockBuilderForModel($model));
 
         $model->publicIncrement('foo', 1);
         $this->assertFalse($model->isDirty());
 
         $model->publicIncrement('foo', 1, ['category' => 1]);
         $this->assertEquals(4, $model->foo);
         $this->assertEquals(1, $model->category);
-        $this->assertTrue($model->isDirty('category'));
+        $this->assertFalse($model->isDirty('category'));
```

Comment:
I changed this behaviour since it's not expected that the property is still dirty after update